userd/autostart.go

```diff
@@ -63,30 +77,81 @@ func expandDesktopFields(in string) string {
 	return string(out)
 }
 
-// Note: consider wrappers/desktop.go if we start parsing more than Exec line
-func findExec(desktopFileContent []byte) (string, error) {
-	scanner := bufio.NewScanner(bytes.NewBuffer(desktopFileContent))
-	execCmd := ""
+type skipDesktopFileError struct {
+	reason string
+}
+
+func (s *skipDesktopFileError) Error() string {
+	return s.reason
+}
+
+func isOneOfIn(of []string, other []string) bool {
+	for _, one := range of {
+		if strutil.ListContains(other, one) {
+			return true
+		}
+	}
+	return false
+}
+
+func loadAutostartDesktopFile(path string) (command string, err error) {
+	f, err := os.Open(path)
```

Comment:
We use an ini file parser in image and partion. We could use it here too, it would be slightly shorter (but not much): http://paste.ubuntu.com/p/5mzGNMWPrZ/ - it is also more liberal (which may be a good thing or a problem depending on POV). Up to you :)

Reply:
I think I'd prefer to avoid loading the whole desktop file into memory.

Reply:
I agree, unless we limit file size in validation.